psygnal/_signal.py

```diff
@@ -852,6 +854,12 @@ def paused(
         finally:
             self.resume(reducer, initial)
 
+    def __getstate__(self) -> dict:
+        """Return dict of current state, for pickle."""
+        d = {slot: getattr(self, slot) for slot in self.__slots__}
+        d.pop("_lock", None)
+        return d
```

Comment:
perhaps!  But since that is _the_ mechanism for providing pickle support (https://docs.python.org/3/library/pickle.html#object.__getstate__) we'd definitely want to make sure that the dict returned by a napari layer was _actually_ fully serializable. (not just "almost" serializable)